data/binding/gen.go

```diff
@@ -521,6 +528,7 @@ type bindValues struct {
 	Format, Since        string
 	SupportsPreferences  bool
 	FromString, ToString string // function names...
+	Custom               bool   // check whether it is a custom type or a built-in one
```

Comment:
"Custom" is pretty vague... perhaps comparable?
It would be better to provide a comparator so that URI (and other future types) are consistent with the rest.

Reply:
> "Custom" is pretty vague... perhaps comparable?

Agree, but I am not sure `comparable` is the right replacement, because as you can see at the end all the types should be comparable, perhaps `BuiltinEquality`?

> It would be better to provide a comparator so that URI (and other future types) are consistent with the rest.

Yes, it was to be my next proposal 😅. Perhaps there could a new interface called `fyne.Equatable`:
```go
type Equatable interface {
     Equal(v Equatable) bool
}
```

Reply:
No need for new types - instead of passing a `bool` then pass in an optional `func(from, to interface{}` so it is handled in this package. URL is easy as you just compare the `String()` versions

Reply:
> No need for new types - instead of passing a bool then pass in an optional func(from, to interface{} so it is handled in this package.

I am bit confused 😅. Is this related to the `Equatable` interface I mentioned or the `Custom` flag?

Reply:
It replaces both

Reply:
Comparator works, no need for tthis to be used for builtin I wouldn't think - nil makes sense.

Reply:
Got it, I will update :)

Reply:
I am not sure how to implement this, you mean `Comparator` should be part of the `bindValues` struct but that struct is only used for template execution. Am I missing something? Maybe I misunderstood you.